zenoh/src/net/routing/hat/router/interests.rs

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use std::sync::Arc;
+
+use zenoh_protocol::{
+    core::WhatAmI,
+    network::{
+        declare::ext,
+        interest::{InterestId, InterestMode, InterestOptions},
+        Declare, DeclareBody, DeclareFinal,
+    },
+};
+use zenoh_sync::get_mut_unchecked;
+
+use super::{
+    face_hat_mut, pubsub::declare_sub_interest, queries::declare_qabl_interest, HatCode, HatFace,
+};
+use crate::net::routing::{
+    dispatcher::{
+        face::FaceState,
+        resource::Resource,
+        tables::{Tables, TablesLock},
+    },
+    hat::{CurrentFutureTrait, HatInterestTrait},
+    RoutingContext,
+};
+
+impl HatInterestTrait for HatCode {
+    fn declare_interest(
+        &self,
+        tables: &mut Tables,
+        _tables_ref: &Arc<TablesLock>,
+        face: &mut Arc<FaceState>,
+        id: InterestId,
+        res: Option<&mut Arc<Resource>>,
+        mode: InterestMode,
+        mut options: InterestOptions,
+    ) {
+        if options.aggregate() && face.whatami == WhatAmI::Peer {
+            tracing::warn!(
+                "Received Interest with aggregate=true from peer {}. Not supported!",
+                face.zid
+            );
+            options -= InterestOptions::AGGREGATE;
```

Comment:
Shouldn't we print an error and return instead? We author the code that might get these options wrong. I think if this happens it's a bug, not a sign that the user is doing something fishy.

Reply:
I just realized that this code seems to predate this pull request. We don't really have to discuss it here.